diag_manager/include/fms_diag_fieldbuff_update.fh

```diff
@@ -0,0 +1,1336 @@
+                  ELSE
+!$OMP CRITICAL
+                     ofb(is-hi:ie-hi,js-hj:je-hj,ks:ke,:,sample) =&
+                     & addwf( ofb(is-hi:ie-hi,js-hj:je-hj,ks:ke,:,sample) , &
+                     & field_d(f1:f2,f3:f4,ks:ke, :) , weight1, pow_value)
+                     !!
+!$OMP END CRITICAL
+                  END IF
+               END IF NDCMP_RKR_4_IF
+!$OMP CRITICAL
+               IF ( .NOT.phys_window ) count_0d = count_0d + weight1
+!$OMP END CRITICAL
+            END IF MISSVAL_PR_3_IF
+         END IF MASK_PR_2_IF ! if mask present
+      END IF MASK_VAR_IF
+
+!$OMP CRITICAL
+      IF ( .NOT.need_compute .AND. .NOT.reduced_k_range ) num_elements = num_elements + &
+        & (ie-is+1)*(je-js+1)*(ke-ks+1)*(le-ls+1)
+      IF ( reduced_k_range ) num_elements = num_elements + &
+        & (ie-is+1)*(je-js+1)*(ker-ksr+1)*(le-ls+1)
+!$OMP END CRITICAL
+
+      succeded = .TRUE.
+      RETURN
+
+    END FUNCTION FMS_DIAG_FBU_PNAME_
+
+
+  !> @brief This code will be used by the preprecessor to generate an implementation
+  !! to the module procudure for the fieldbuff_copy_fieldvals interface. The
+  !! generated function is a wrapper calling 4D field/5D buffer version of the same.
+    FUNCTION FMS_DIAG_FBCF_3D_PNAME_ (ofield_cfg, ofield_index_cfg, field, sample, ofb, &
+      & bbounds, count_0d, mask,  missvalue,  &
+      & l_start, l_end, err_msg,  err_msg_local) result( succeded )
+         TYPE(fms_diag_outfield_type), INTENT(in) :: ofield_cfg   !< The ofield_cfg object
+         TYPE(fms_diag_outfield_index_type) , INTENT(in) :: ofield_index_cfg    !< The ofield_index_cfg object
+         FMS_DIAG_FBU_DATA_TYPE_ , CONTIGUOUS,  DIMENSION(:,:,:),INTENT(in),target:: field !< The field value array.
+         INTEGER, INTENT(in) :: sample   !< index along the diurnal time axis
+         FMS_DIAG_FBU_DATA_TYPE_ , ALLOCATABLE, DIMENSION(:,:,:,:),INTENT(inout),target:: ofb !<The Output Field Buffer
+         TYPE(fms_diag_ibounds_type), INTENT(inout) :: bbounds
+         FMS_DIAG_FBU_DATA_TYPE_ , INTENT(inout) :: count_0d   !> Normally the member of the buffer of same name,
+         LOGICAL, CONTIGUOUS, DIMENSION(:,:,:), INTENT(in), OPTIONAL, target :: mask
+         FMS_DIAG_FBU_DATA_TYPE_ , INTENT(in) :: missvalue !> buffer may be set to this value where mask is false.
+         INTEGER, DIMENSION(3), INTENT(in)  :: l_start   !< local start indices on spatial axes for regional output
+         INTEGER, DIMENSION(3), INTENT(in)  :: l_end   !< local end indices on spatial for regional output
+         CHARACTER(len=*), INTENT(inout), OPTIONAL :: err_msg
+         CHARACTER(len=256), INTENT(inout) :: err_msg_local
+
+        !! For pointer bounds remapping
+        FMS_DIAG_FBU_DATA_TYPE_ , pointer, DIMENSION(:,:,:,:) :: field_ptr => null() !<
+        FMS_DIAG_FBU_DATA_TYPE_ , pointer, DIMENSION(:,:,:,:,:) :: ofb_ptr => null() !<
+        LOGICAL , pointer, DIMENSION(:,:,:,:) :: mask_ptr => null() !<
```

Comment:
These look like you wanted to add a comment, but didnt.

Reply:
Added comment,